src/Configuration/src/Encryption/Decryption/ConfigServerEncryptionResolverExtensions.cs

```diff
@@ -0,0 +1,208 @@
+    /// and provide encryption resolution for the configuration. Typically, you will want to add this configuration source as the last one so that you wrap
+    /// all of the applications' configuration sources with encryption resolution.
+    /// </summary>
+    /// <param name="hostBuilder">
+    /// The host builder.
+    /// </param>
+    /// <returns>
+    /// The provided host builder.
+    /// </returns>
+    public static IWebHostBuilder AddConfigServerEncryptionResolver(this IWebHostBuilder hostBuilder)
+    {
+        return AddConfigServerEncryptionResolver(hostBuilder, NullLoggerFactory.Instance);
+    }
+
+    /// <summary>
+    /// Adds a encryption resolver configuration source to the <see cref="ConfigurationBuilder" />. The encryption resolver source will capture and wrap all
+    /// the existing sources <see cref="IConfigurationSource" /> contained in the builder.  The newly created source will then replace the existing sources
+    /// and provide encryption resolution for the configuration. Typically, you will want to add this configuration source as the last one so that you wrap
+    /// all of the applications' configuration sources with encryption resolution.
+    /// </summary>
+    /// <param name="hostBuilder">
+    /// The host builder.
+    /// </param>
+    /// <param name="loggerFactory">
+    /// Used for internal logging. Pass <see cref="NullLoggerFactory.Instance" /> to disable logging.
+    /// </param>
+    /// <returns>
+    /// The provided host builder.
+    /// </returns>
+    public static IWebHostBuilder AddConfigServerEncryptionResolver(this IWebHostBuilder hostBuilder, ILoggerFactory loggerFactory)
+    {
+        ArgumentGuard.NotNull(hostBuilder);
+        ArgumentGuard.NotNull(loggerFactory);
+
+        return hostBuilder.ConfigureAppConfiguration((context, builder) =>
+        {
+            ITextDecryptor textDecryptor = GetTextDecryptor(context.Configuration);
+            builder.AddEncryptionResolver(loggerFactory, textDecryptor);
+        });
+    }
+
+    /// <summary>
+    /// Adds a encryption resolver configuration source to the <see cref="ConfigurationBuilder" />. The encryption resolver source will capture and wrap all
+    /// the existing sources <see cref="IConfigurationSource" /> contained in the builder.  The newly created source will then replace the existing sources
+    /// and provide encryption resolution for the configuration. Typically, you will want to add this configuration source as the last one so that you wrap
+    /// all of the applications configuration sources with encryption resolution.
+    /// </summary>
+    /// <param name="hostBuilder">
+    /// The host builder.
+    /// </param>
+    /// <returns>
+    /// The provided host builder.
+    /// </returns>
+    public static IHostBuilder AddConfigServerEncryptionResolver(this IHostBuilder hostBuilder)
```

Comment:
Should these be simply called AddEncryptionResolver since this is now separate from ConfigServer?

Reply:
Added integration test for combined use of PlaceholderResolver and EncryptionResolver